java/dataset/src/main/java/org/apache/arrow/dataset/scanner/ScanOptions.java

```diff
@@ -25,8 +26,9 @@
  * Options used during scanning.
  */
 public class ScanOptions {
-  private final Optional<String[]> columns;
+  private final Optional<String[]> columnsSubset;
   private final long batchSize;
+  private Optional<ByteBuffer> columnsProduceOrFilter;
```

Comment:
I'd suggest we still expose a simple unbound string in the interface, but leaving the construction (ser/deser) process in the dataset jar itself.

I mean it's good to use subtrait as underlying implementation, but maybe we should just keep the interfaces simple.

Reply:
Hey @lidavidm, what's your take on this?

Reply:
The string parsing can and should be kept separate. The actual interface should reflect Substrait. (We could add a convenience method to set the filter via a string, which gets immediately parsed, for instance.)